openretina/neuron_data_io.py

```diff
@@ -452,7 +468,120 @@ def _upsample_triggertimes(stim_empirical_duration, stim_theoretical_duration, t
     return upsampled_triggertimes
 
 
-def make_final_responses(data_dict: dict, response_type: Literal["natural", "chirp", "mb"] = "natural"):
+def _apply_mask_to_field(data_dict, field, mask):
+    """
+    Apply a mask to a specific field in a data dictionary.
+
+    Args:
+        data_dict (dict): A dictionary containing data fields.
+        field (str): The field in the data dictionary to apply the mask to.
+        mask (np.ndarray): The mask to apply to the field.
+
+    Returns:
+        None
+
+    Raises:
+        IndexError: If the mask index is out of bounds for the field data.
+
+    Examples:
+        _apply_mask_to_field(data_dict, 'field_name', mask)"""
+
+    for key in data_dict[field].keys():
+        if key in ["roi_mask", "roi_coords"]:
+            continue
+        if isinstance(data_dict[field][key], np.ndarray) and len(data_dict[field][key]) > 0:
+            if len(data_dict[field][key].shape) == 1:
+                data_dict[field][key] = data_dict[field][key][mask]
+            elif len(data_dict[field][key].shape) == 2:
+                if data_dict[field][key].shape[0] == mask.shape[0]:
+                    data_dict[field][key] = data_dict[field][key][mask, :]
+                else:
+                    data_dict[field][key] = data_dict[field][key][:, mask]
+            else:
+                raise IndexError(f"Index out of bounds for field {field} and key {key}.")
+
+
+def _apply_qi_mask(data_dict, qi_types: list[str], qi_thresholds: list[float], logic="or"):
+    """
+    Applies quality thresholds as a mask to the data dictionary.
+
+    Args:
+        data_dict (dict): The data dictionary.
+        qi_types (list): List of quality index types.
+        qi_threshold (list): List of quality index thresholds.
+        logic (str): The logic to combine different qi_types. Can be 'and' or 'or'. Default is 'and'.
+
+    Returns:
+        dict: The updated data dictionary.
+    """
+    new_data_dict = deepcopy(data_dict)
+
+    if logic not in ["and", "or"]:
+        raise ValueError("logic must be either 'and' or 'or'")
+    assert len(qi_types) == len(qi_thresholds), "qi_types and qi_thresholds must have the same length"
+
+    for field in new_data_dict.keys():
+        masks = [
+            new_data_dict[field][f"{qi_type}_qi"] >= qi_threshold
+            for qi_type, qi_threshold in zip(qi_types, qi_thresholds)
```

Comment:
```suggestion
            for qi_type, qi_threshold in zip(qi_types, qi_thresholds, strict=True)
```
If you use this you probably don't need the assert above.